packages/user-service/src/cron.ts

```diff
@@ -0,0 +1,46 @@
+import * as _ from 'lodash';
+import * as async from 'async';
+import { User } from './schema';
+import moment from 'moment';
+import { UserGroupAPIHelper } from './helpers';
+/**
+ * @class UserSyncCron
+ */
+export class UserSyncCron {
+  public syncUsers() {
+    User.find().then((userInfo: any[]) => {
+      if (userInfo.length) {
+        let rhatUUIDs: string[] = [];
+        rhatUUIDs = userInfo.map((user: UserType) => user.rhatUUID);
+        if (rhatUUIDs.length) {
+          async.map(rhatUUIDs, rhatUUID => {
+            UserGroupAPIHelper.getProfilesBy(`(rhatUUID=${rhatUUID})`)
```

Comment:
Also, the same result can be achieved using regular ES6 map method.

```suggestion
          rhatUUIDs.map(rhatUUID => {
            return UserGroupAPIHelper.getProfilesBy(`(rhatUUID=${rhatUUID})`)
```

Reply:
And that will give you proper control over error handling.

Reply:
updated as suggested